src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -123,15 +123,13 @@ public virtual SdkResult ResolveSdk(int submissionId, SdkReference sdk, LoggingC
                     // to load NuGet assemblies at runtime which could fail if the user is not running installed MSBuild.  Rather
                     // than give them a generic error, we want to give a more specific message.  This exception cannot be caught by
                     // the resolver itself because it is usually thrown before the class is loaded
-                    // MSB4243: The NuGet-based SDK resolver failed to run because NuGet assemblies could not be located.  Check your installation of MSBuild or set the environment variable "{0}" to the folder that contains the required NuGet assemblies. {1}
-                    loggingContext.LogWarning(null, new BuildEventFileInfo(sdkReferenceLocation), "CouldNotRunNuGetSdkResolver", MSBuildConstants.NuGetAssemblyPathEnvironmentVariableName, e.Message);
-                    continue;
+                    // The NuGet-based SDK resolver failed to run because NuGet assemblies could not be located.  Check your installation of MSBuild or set the environment variable "{0}" to the folder that contains the required NuGet assemblies. {1}
+                    throw new SdkResolverException("CouldNotRunNuGetSdkResolver", sdkResolver, sdk, e, MSBuildConstants.NuGetAssemblyPathEnvironmentVariableName, e.Message);
```

Comment:
So I know this is what was here before, but `Exception.Message` is a bit of a code smell since it hides information (specifically the stack at the point of failure). Should that just be
```suggestion
                    throw new SdkResolverException("CouldNotRunNuGetSdkResolver", sdkResolver, sdk, e, MSBuildConstants.NuGetAssemblyPathEnvironmentVariableName, e.ToString());
```

?

Reply:
It looks like `ProjectErrorUtilities.ThrowInvalidProject` prepends the error code to every line of the stack trace. Looks a little messy, is this still preferred?

```
Build FAILED.

C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242: SDK Resolver Failure: "The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed while attempting to resolve the SDK "Microsoft.NET.Sdk". Exception: "System.Text.Json.JsonReaderException: 's' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8.
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes) in System.Text.Json.dll:token 0x60000de+0x0
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenUntilAfterAllCommentsAreSkipped(Byte marker) in System.Text.Json.dll:token 0x6000315+0x1fc
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextToken(Byte marker) in System.Text.Json.dll:token 0x6000311+0x3c
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenOrRollback(Byte marker) in System.Text.Json.dll:token 0x6000310+0x24
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ReadSingleSegment() in System.Text.Json.dll:token 0x60002fc+0x113
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.Read() in System.Text.Json.dll:token 0x60002e8+0x8
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.Utf8JsonStreamReader.Read() in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x60000ef+0x12f
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinition(WorkloadId id, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000082+0x2ff
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinitions(Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600007c+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String id, String informationalPath, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000079+0x124
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String manifestId, Stream manifestStream, String informationalPath) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000084+0x23
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvider) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600009e+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000099+0x8a
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.CachingWorkloadResolver.Resolve(String sdkReferenceName, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x6000019+0x5b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.WorkloadSdkResolver.Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x600001c+0x50
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.SdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\SdkResolverService.cs:line 118""
    0 Warning(s)
    1 Error(s)
```